### PR TITLE
Added support for replicates. Test needed.

### PR DESCRIPTION
**Pull request by charles-cowart**
This is how GenPrepFileJob will use the new functionality in metapool to transparently handle replicates. The code is here but I need to write a good test.

### CODE REVIEW

**Comment by charles-cowart** [comment]
This PR depends on an update to metapool (https://github.com/biocore/metagenomics_pooling_notebook/pull/134). 134 is currently failing due to a Qiita dependency. This issue will be looked into shortly.

**Comment by charles-cowart** [comment]
@antgonza Just a reminder this should ideally be reviewed after the changes to metapool are approved and merged and the url in setup.py is reverted to normal.

File: sequence_processing_pipeline/GenPrepFileJob.py [hunk 1]
```diff
@@ -68,13 +74,65 @@ def __init__(self, run_dir, convert_job_path, qc_job_path, output_path,
...
+        w/a unique name.
+        :param demuxed:
+        :return:
+        '''
+        count = 0
+        results = []
+        for replicate in demuxed:
+            count += 1
+            if self.is_amplicon:
+                fp = join(self.output_path, f"sheet{count}.txt")
```
Comment (antgonza) [inline sequence_processing_pipeline/GenPrepFileJob.py:120]: would an underscore be helpful here? Like sheet_X.txt

**Review by antgonza — CHANGES_REQUESTED** [review]

**Comment by charles-cowart** [comment]
@antgonza ready for re-review!

**Review by antgonza — CHANGES_REQUESTED** [review]
Sorry missed a line in my last review, I think this is it.

File: sequence_processing_pipeline/GenPrepFileJob.py [hunk 2]
```diff
@@ -68,13 +74,65 @@ def __init__(self, run_dir, convert_job_path, qc_job_path, output_path,
...
+        '''
+        Saves the new plate-replicate-specific sample-sheet or pre-prep file
+        w/a unique name.
+        :param demuxed:
+        :return:
+        '''
+        count = 0
+        results = []
+        for replicate in demuxed:
+            count += 1
```
[the remark this answers is not in the document]
Comment (charles-cowart) [reply sequence_processing_pipeline/GenPrepFileJob.py:118]: ah good catch! Sometimes I still think in C I guess. 😆

**Pull request merged**